### ISSUE TITLE
Blob gpu_0/keypoint_visible_locations_int32 not in the workspace

### ISSUE THREAD

**Issue by lilichu** [facebookresearch/Detectron#679]
hi! I add a branch(copy the keypoint head) in keypoint-faster_rcnn to class the keypoint(true-corner or false-corner). 
After I has build the net structure, when runing, I encounter the issue such as 
`Blob gpu_0/keypoint_visible_locations_int32 not in the workspace`
I guess I should do something else other than building the net. how should I do? thanks!
the more information of the issue:
```
INFO net.py:  59: Loading weights from: /tmp/detectron-download-cache/ImageNetPretrained/MSRA/R-50.pkl
INFO net.py:  88: fpn_inner_res5_2_sum_w not found
INFO net.py:  88: fpn_inner_res5_2_sum_b not found
INFO net.py:  88: fpn_inner_res4_5_sum_lateral_w not found
INFO net.py:  88: fpn_inner_res4_5_sum_lateral_b not found
INFO net.py:  88: fpn_inner_res3_3_sum_lateral_w not found
INFO net.py:  88: fpn_inner_res3_3_sum_lateral_b not found
INFO net.py:  88: fpn_inner_res2_2_sum_lateral_w not found
INFO net.py:  88: fpn_inner_res2_2_sum_lateral_b not found
INFO net.py:  88: fpn_res5_2_sum_w not found
INFO net.py:  88: fpn_res5_2_sum_b not found
INFO net.py:  88: fpn_res4_5_sum_w not found
INFO net.py:  88: fpn_res4_5_sum_b not found
INFO net.py:  88: fpn_res3_3_sum_w not found
INFO net.py:  88: fpn_res3_3_sum_b not found
INFO net.py:  88: fpn_res2_2_sum_w not found
INFO net.py:  88: fpn_res2_2_sum_b not found
INFO net.py:  88: conv_rpn_fpn2_w not found
INFO net.py:  88: conv_rpn_fpn2_b not found
INFO net.py:  88: rpn_cls_logits_fpn2_w not found
INFO net.py:  88: rpn_cls_logits_fpn2_b not found
INFO net.py:  88: rpn_bbox_pred_fpn2_w not found
INFO net.py:  88: rpn_bbox_pred_fpn2_b not found
INFO net.py:  88: fc6_w not found
INFO net.py:  88: fc6_b not found
INFO net.py:  88: fc7_w not found
INFO net.py:  88: fc7_b not found
INFO net.py:  88: cls_score_w not found
INFO net.py:  88: cls_score_b not found
INFO net.py:  88: bbox_pred_w not found
INFO net.py:  88: bbox_pred_b not found
INFO net.py:  88: conv_fcn1_w not found
INFO net.py:  88: conv_fcn1_b not found
INFO net.py:  88: conv_fcn2_w not found
INFO net.py:  88: conv_fcn2_b not found
INFO net.py:  88: conv_fcn3_w not found
INFO net.py:  88: conv_fcn3_b not found
INFO net.py:  88: conv_fcn4_w not found
INFO net.py:  88: conv_fcn4_b not found
INFO net.py:  88: conv_fcn5_w not found
INFO net.py:  88: conv_fcn5_b not found
INFO net.py:  88: conv_fcn6_w not found
INFO net.py:  88: conv_fcn6_b not found
INFO net.py:  88: conv_fcn7_w not found
INFO net.py:  88: conv_fcn7_b not found
INFO net.py:  88: conv_fcn8_w not found
INFO net.py:  88: conv_fcn8_b not found
INFO net.py:  88: kps_score_lowres_w not found
INFO net.py:  88: kps_score_lowres_b not found
INFO net.py:  88: kps_score_w not found
INFO net.py:  88: kps_score_b not found
INFO net.py:  88: visible_conv_fcn1_w not found
INFO net.py:  88: visible_conv_fcn1_b not found
INFO net.py:  88: visible_conv_fcn2_w not found
INFO net.py:  88: visible_conv_fcn2_b not found
INFO net.py:  88: visible_conv_fcn3_w not found
INFO net.py:  88: visible_conv_fcn3_b not found
INFO net.py:  88: visible_conv_fcn4_w not found
INFO net.py:  88: visible_conv_fcn4_b not found
INFO net.py:  88: visible_conv_fcn5_w not found
INFO net.py:  88: visible_conv_fcn5_b not found
INFO net.py:  88: visible_conv_fcn6_w not found
INFO net.py:  88: visible_conv_fcn6_b not found
INFO net.py:  88: visible_conv_fcn7_w not found
INFO net.py:  88: visible_conv_fcn7_b not found
INFO net.py:  88: visible_conv_fcn8_w not found
INFO net.py:  88: visible_conv_fcn8_b not found
INFO net.py:  88: kps_visible_score_lowres_w not found
INFO net.py:  88: kps_visible_score_lowres_b not found
INFO net.py:  88: kps_visible_score_w not found
INFO net.py:  88: kps_visible_score_b not found
W0924 21:14:21.302081 21491 workspace.cc:174] Blob gpu_0/keypoint_visible_locations_int32 not in the workspace.
WARNING workspace.py: 185: Original python traceback for operator `328` in network `generalized_rcnn` in exception above (most recent call last):
WARNING workspace.py: 190:   File "/home/user/pycharm-2018.1.3/helpers/pydev/pydevd.py", line 1664, in <module>
WARNING workspace.py: 190:   File "/home/user/pycharm-2018.1.3/helpers/pydev/pydevd.py", line 1658, in main
WARNING workspace.py: 190:   File "/home/user/pycharm-2018.1.3/helpers/pydev/pydevd.py", line 1068, in run
WARNING workspace.py: 190:   File "/home/user/backup/detectron/tools/train_net.py", line 156, in <module>
WARNING workspace.py: 190:   File "/home/user/backup/detectron/tools/train_net.py", line 111, in main
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/utils/train.py", line 53, in train_model
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/utils/train.py", line 137, in create_model
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/model_builder.py", line 124, in create
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/model_builder.py", line 89, in generalized_rcnn
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/model_builder.py", line 235, in build_generic_detection_model
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/optimizer.py", line 40, in build_data_parallel_model
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/optimizer.py", line 63, in _build_forward_graph
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/model_builder.py", line 223, in _single_gpu_build_func
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/model_builder.py", line 355, in _add_roi_keypoint_visible_head
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/modeling/keypoint_rcnn_heads.py", line 261, in add_keypoint_visible_losses
WARNING workspace.py: 190:   File "/home/user/backup/detectron/detectron/utils/blob.py", line 165, in get_loss_gradients
```

**Comment by ir413** [comment]
Hi @lilichu, you need to ensure that the input blobs for a Caffe2 op are present in the workspace when that operator is run. The error you're getting is due to the `keypoint_visible_locations_int32` blob not being present in the workspace when the operator that uses it is run. You should ensure that the blob is loaded before running the operator.

From the name of the blob, it seems that it represents ground truth labels for your task? If so, it may be instructive to see how ground truth labels are loaded for the existing tasks (e.g. see [here](https://github.com/facebookresearch/Detectron/blob/master/detectron/roi_data/keypoint_rcnn.py#L90) for keypoints).

**Comment by lilichu** [comment]
@ir413 thanks for your reply.  well, I have changed the `keypoints_to_heatmap_labels` to get `visible ` and add the biobs such as :

```
    heats, weights, visible = keypoint_utils.keypoints_to_heatmap_labels(
        sampled_keypoints, sampled_fg_rois
    )
......
blobs['keypoint_visible_locations_int32'] = visible.astype(np.int32, copy=False)
```
what esle should I do? Thanks!

**Comment by ir413** [comment]
@lilichu you should also add the new blob name [here](https://github.com/facebookresearch/Detectron/blob/master/detectron/roi_data/fast_rcnn.py#L71).

**Comment by lilichu** [comment]
@ir413  thanks！

**Comment by mamunir** [comment]
Blob gpu_0/fpn_res5_2_sum_1 not in the workspace

I am facing problem mentioned above. Any clue, where to put it in code so that this error removes?

@lilichu @ir413
